src/App.tsx

```diff
@@ -15,71 +20,124 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
-
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
-
-        <tr data-cy="Good">
-          <td>
+export class App extends React.Component {
+  state = {
+    selectedGood: 'Jam',
+  };
+
+  reset = () => {
+    goods.forEach(good => {
+      const element:HTMLElement | null = document.getElementById(`${good}`);
+
+      if (element) {
+        element.classList.remove('has-background-success-light');
+        const button:HTMLElement | null = element.querySelector('.button');
+
+        if (button && button.classList.contains('is-info')) {
+          button.classList.remove('is-info');
+          button.setAttribute('data-cy', 'AddButton');
+          button.textContent = '+';
+        }
+      }
+    });
+
+    this.setState({ selectedGood: null });
+  };
+
+  handleClick = (good: string) => {
```

Comment:
this logic looks even more complicated as removing item, this logic should only update selectedGood in state, not doing any manipulation on selectors or loops, they are redundant, try this one:

```
 selectedItem = (good: string) => {
    this.setState({ selectedGood: good });
 };
```